### ISSUE TITLE
[Bug] Broken plugin config files breaks plugins

### ISSUE THREAD

**Issue by Skamt** [BetterDiscord/BetterDiscord#1794]
### Before Reporting

- [X] I found no existing issues matching my bug
- [X] My issue is not caused by a plugin or theme

### Describe the Bug

Messed up plugin config files will cause plugins to not load because of the way BD reads them.
refer to https://github.com/Skamt/BDAddons/issues/59

### Expected Behavior

Do not break plugins if config files can't be loaded for whatever reason

### Steps To Reproduce

1. manually edit a plugin config file and make it incompatible with JSON *(empty file would work)*
2. open discord

*This can happen without user meddling*

### Operating System

Windows

### Discord Version

Stable

### BetterDiscord Version

1.11.0

**Comment by Huderon** [comment]
Do you have any ideas how you actually want this to be resolved? I'm probably wrong, but AFAIK all that really needs to happen is wrap [this line](https://github.com/BetterDiscord/BetterDiscord/blob/3de6da41282d23f008e3f825b2b2051537dbc979/renderer/src/modules/datastore.js#L109) in a try...catch. Deleting the file when it catches doesn't seem right, though, and I can't imagine asking end users to resolve the conflict on their own would go well. Showing a popout and giving the option to either delete or resolve it on their own seems like a decent middle ground to me, but I think that'd also confuse a fair number of users

**Comment by Skamt** [comment]
Yes, a try...catch around `JSON.parse` because it just throws when it can't parse for any reason and if so pretend the config file is empty and return a brand new empty object.

another soulution would be to use a json parsing library that is not as strict.

**Comment by Huderon** [comment]
Yeah, returning an empty object is fine, but what do you do with the existing config? Just delete it? The corruption needs to be resolved somehow, right? or it'll just keep saving to that corrupt file and never be able to load

**Comment by Skamt** [comment]
there's only two case where the parsing would fail, One is a completely corrupted file in this case there is no 'existing config' just return an empty object.

The other one is json syntax error which we can try to recover using a less strict json library (one that doesn't freakout over a trailing comma) otherwise return an empty object

**Comment by Huderon** [comment]
I might be a dummy, but I believe it will only try to parse the file after confirming it exists, so there's definitely a config there, corrupted or no. Returning an empty object if it fails is fine, but then the next time you call `Data.save` it's going to attempt to save to that corrupted file, and every time you call `Data.load` it's going to attempt to parse that corrupted file again. Something needs to be done to clean up that file. Deleting it seems fine to me, but I imagine that'd bother quite a few people.

I'm a fan of trailing commas myself, but I can't imagine we'd get the ok to make that change. If we're using JSON we should probably use it correctly, regardless
